homeassistant/components/downloader/__init__.py

```diff
@@ -1,51 +1,37 @@
 """Support for functionality to download files."""
+import asyncio
 from http import HTTPStatus
-import logging
 import os
 import re
 import threading
+import time
 
 import requests
 import voluptuous as vol
 
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant, ServiceCall
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.typing import ConfigType
+from homeassistant.helpers.service import async_register_admin_service
 from homeassistant.util import raise_if_invalid_filename, raise_if_invalid_path
 
-_LOGGER = logging.getLogger(__name__)
-
-ATTR_FILENAME = "filename"
-ATTR_SUBDIR = "subdir"
-ATTR_URL = "url"
-ATTR_OVERWRITE = "overwrite"
-
-CONF_DOWNLOAD_DIR = "download_dir"
-
-DOMAIN = "downloader"
-DOWNLOAD_FAILED_EVENT = "download_failed"
-DOWNLOAD_COMPLETED_EVENT = "download_completed"
-
-SERVICE_DOWNLOAD_FILE = "download_file"
-
-SERVICE_DOWNLOAD_FILE_SCHEMA = vol.Schema(
-    {
-        vol.Required(ATTR_URL): cv.url,
-        vol.Optional(ATTR_SUBDIR): cv.string,
-        vol.Optional(ATTR_FILENAME): cv.string,
-        vol.Optional(ATTR_OVERWRITE, default=False): cv.boolean,
-    }
-)
-
-CONFIG_SCHEMA = vol.Schema(
-    {DOMAIN: vol.Schema({vol.Required(CONF_DOWNLOAD_DIR): cv.string})},
-    extra=vol.ALLOW_EXTRA,
+from .const import (
+    _LOGGER,
+    ATTR_FILENAME,
+    ATTR_OVERWRITE,
+    ATTR_SUBDIR,
+    ATTR_URL,
+    CONF_DOWNLOAD_DIR,
+    DOMAIN,
+    DOWNLOAD_COMPLETED_EVENT,
+    DOWNLOAD_FAILED_EVENT,
+    SERVICE_DOWNLOAD_FILE,
 )
```

Comment:
We can't remove this until the import flow is removed.

Reply:
It's not removed, but replaced to [here](https://github.com/home-assistant/core/pull/98722/files/020523912a6e68fe348110a638e7e0b0c0d604b1#diff-6fb93748daa489e86a67ffaccba74185628757d8eab65f413f4e56d0314bb53dR52). It now registers directly.